File: deletebv.py
import oci
from datetime import datetime,timedelta

config = oci.config.from_file("~/.oci/config","DEFAULT")

# Initialize service client with default config file
core_client = oci.core.BlockstorageClient(config)
ocicompartment="<<enter compartment id>>"

# Send the request to service, some parameters are not required, see API
# LIST VOLUMES
list_volumes_response = core_client.list_volumes(
    compartment_id=ocicompartment,
    availability_domain="CISk:US-ASHBURN-AD-1",
    limit=903,
    sort_by="TIMECREATED",
    sort_order="DESC",
)

# List all the volumes
#print(list_volumes_response.data)
l = len(list_volumes_response.data)
# print(l)

# For each volume id - delete the volume backup
for i in range(0,l):
    print("Backup for the Volume -" + list_volumes_response.data[i].display_name + " will be deleted")
    list_volume_backups_response = core_client.list_volume_backups(
        compartment_id=ocicompartment,
        volume_id=list_volumes_response.data[i].id,
        limit=156,
        sort_by="TIMECREATED",
        sort_order="DESC",
        lifecycle_state="AVAILABLE"
    )
    # LIST VOLUME BACKUP FOR THE VOLUME ID
    print(list_volume_backups_response.data)
    volBackupsLen= len(list_volume_backups_response.data)
    j=0
    # Delete backups greater than 30 days
    for j in range(0,volBackupsLen):
        # print("Date - "+str(list_volume_backups_response.data[j].time_created))
        # print(backuptime)
        # print(timebefore30days)
        timebefore30days = datetime.now() - timedelta(days=30)
        volbackuptime = datetime.strptime(str(list_volume_backups_response.data[j].time_created),
                                       '%Y-%m-%d %H:%M:%S.%f+00:00')

        if timebefore30days > volbackuptime:
            print("Delete backups greater than 30 days")
            print("Deleting volume -" + list_volume_backups_response.data[j].display_name)
            delete_volume_backup_response = core_client.delete_volume_backup(
                volume_backup_id=list_volume_backups_response.data[j].id
            )
            # Get the data from response
            print(delete_volume_backup_response.headers)
        #
        # else:
        #     print("No backups or All backups are created less than 30 days")

File: deletebv.java
import com.oracle.bmc.ConfigFileReader;
import com.oracle.bmc.auth.AuthenticationDetailsProvider;
import com.oracle.bmc.auth.ConfigFileAuthenticationDetailsProvider;
import com.oracle.bmc.core.BlockstorageClient;
import com.oracle.bmc.core.model.*;
import com.oracle.bmc.core.requests.*;
import com.oracle.bmc.core.responses.*;
import net.minidev.json.JSONUtil;

import javax.xml.bind.SchemaOutputResolver;
import java.util.*;


public class VolumeBackups {
    public static void main(String[] args) throws Exception {

        final ConfigFileReader.ConfigFile configFile = ConfigFileReader.parseDefault();
        final AuthenticationDetailsProvider provider = new ConfigFileAuthenticationDetailsProvider(configFile);

        BlockstorageClient client = new BlockstorageClient(provider);

        ListVolumeBackupsRequest listVolumeBackupsRequest = ListVolumeBackupsRequest.builder()
                .compartmentId("")
                .limit(1000)
                .sortBy(ListVolumeBackupsRequest.SortBy.Timecreated)
                .sortOrder(ListVolumeBackupsRequest.SortOrder.Desc)
                .lifecycleState(VolumeBackup.LifecycleState.Available).build();

        ListVolumeBackupsResponse response = client.listVolumeBackups(listVolumeBackupsRequest);

        List<VolumeBackup> list =  response.getItems();
        for(VolumeBackup v:list){
            System.out.println("Volume Name: " + v.getDisplayName());
            System.out.println("Time Created: " + v.getTimeCreated());
            Date date = new Date();
            long diffInMillies = date.getTime() - v.getTimeCreated().getTime();
            long difference_In_Days = (diffInMillies / (1000 * 60 * 60 * 24)) % 365;
            System.out.println("Difference in Days: " + difference_In_Days);
            if(difference_In_Days > 30)
            {
                System.out.println("delete the volume");
                deleteVolume(client, v.getId());
            }
            else
                System.out.println("keep the volume");
        }
    }

    private static void deleteVolume(BlockstorageClient client, String id){

        DeleteVolumeBackupRequest deleteVolumeBackupRequest = DeleteVolumeBackupRequest.builder()
                .volumeBackupId(id).build();

        DeleteVolumeBackupResponse response = client.deleteVolumeBackup(deleteVolumeBackupRequest);
    }


}
